documentation/src/main/asciidoc/topics/proc_configuring_cache_tracing.adoc

```diff
@@ -0,0 +1,71 @@
+[id='tracing_cache']
+= Configure tracing at cache level
+
+Once the tracing is link:#configuring-tracing_{context}[configured at server level],
+it will be automatically enabled by default for all caches.
+A cache configuration level of tracing allows on the other hand to enable or disable it at cache level and at runtime.
+
+[discrete]
+== Tracing categories
+
+Several categories are potentially traced:
+
+* Container. That are all the main cache operations, such as replace, put, clear, getForReplace, remove operations and size. With the exception of all the getting operation.
+* Cluster. All the operations involved at clustering level, e.g., to align a replicas.
+* X-Site. All the operations involved at cross-site level, e.g, to align a remote site.
+* Persistence. All the operations involving persistence (cache store).
+
+Each of category can be enabled/disabled at start time or runtime using the related tracing configuration attributes.
+By default only the container category is enabled.
+
+There is also the `Security` category, to trace security audit events.
+This category is configured globally, not only at cache level, since their events can have different scopes (cache, container or server),
+non only cache scope.
+
+[discrete]
+== Enable/disable tracing for a given cache
+{brandname} applies the tracing configuration globally to all caches.
+
+.XML
+[source,xml,options="nowrap",subs=attributes+,role="primary"]
+----
+include::xml/tracing_cache_config.xml[]
+----
+
+.JSON
+[source,json,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::json/tracing_cache_config.json[]
+----
+
+.YAML
+[source,yaml,options="nowrap",subs=attributes+,role="secondary"]
+----
+include::yaml/tracing_cache_config.yaml[]
+----
+
+[discrete]
+== Enable/disable tracing at runtime
```

Comment:
Should we include an example for enabled/disabling a category?

Reply:
Also have you actually tested this? I still don't see how this is going to work tbh.. as I mentioned a few times before the code I would normally expect to see for updating values at runtime without a full server restart are not present. Normally this is done via a listener on the configuration element.

https://github.com/infinispan/infinispan/blob/main/commons/all/src/main/java/org/infinispan/commons/configuration/attributes/Attribute.java#L111

Reply:
IIUC, instead of using the listener, each traced operation retrieves the local configuration and checks if the tracing is enabled. It relies on the listener installed on the configuration cache to update the local version. I think it would need to use the listener if the configuration was stored in a variable.

But thinking about it again, it might work for the JGroups requests because it uses the global component. But it wouldn't update the persistence tracing, as it only checks when starting a cache which one to inject.

Reply:
Correct, but the code before was storing the configuration value for being enabled in a variable.. but it has changed so many times now I don't even remember which was which.

Basically if every operation would check for tracing was enabled every time and was implanted in the code then yes it could work. The component replacement should look solely at the global configuration value that is not runtime updateable.

Reply:
Thank you for the check.